pkg/maps/mapregistry/cell.go

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package mapregistry
+
+import (
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/maps"
+	"github.com/cilium/cilium/pkg/maps/authmap"
+	"github.com/cilium/cilium/pkg/maps/configmap"
+)
+
+// Cell provides BPF Map registry, which contains all BPF Maps which have been registered.
+// Currently, its main purpose is to depend on the initialization of the maps when starting Cilium.
+var Cell = cell.Module(
+	"map-registry",
+	"BPF Map Registry",
+
+	// Provides the auth.Map which contains the authentication state between Cilium security identities.
+	authmap.Cell,
+
+	// ConfigMap stores runtime configuration state for the Cilium datapath.
+	configmap.Cell,
+
+	cell.Provide(newMapRegistry),
+)
+
+func newMapRegistry(params mapRegistryParams) *MapRegistry {
+	return &MapRegistry{
+		BpfMaps: params.BpfMaps,
+	}
+}
+
+// MapRegistry contains all BPF Maps which have been registered.
+type MapRegistry struct {
```

Comment:
I would just add to datapathParams:
```
Maps []maps.BpfMap `group:"bpf-maps"`
```

This wrapper seems like an unnecessary indirection.